### PR TITLE
Allow custom transports

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,6 +119,26 @@ The transport can be configured with the `board_microros_transport = <transport>
     set_microros_native_ethernet_transports(local_mac, local_ip, agent_ip, agent_port);
     ```
 
+  - `custom`
```
Comment (Acuadros95) [inline README.md:122]: We should move this to this section:
https://github.com/micro-ROS/micro_ros_platformio/blob/main/README.md#custom-transport

Comment (pablogs9) [reply README.md:122]: Maybe we should rename this section as how to extend available library transports because concepts are similar but not the same

**Comment by pablogs9** [comment]
What is the rationale of a2a6a8d6d1b424137d26c463f8283b0c83fab5cb?

**Comment by Acuadros95** [comment]
@pablogs9 https://github.com/micro-ROS/micro_ros_platformio/commit/a2a6a8d6d1b424137d26c463f8283b0c83fab5cb is related to this comment: https://github.com/micro-ROS/micro_ros_platformio/discussions/2#discussioncomment-2639526

**Comment by pablogs9** [comment]
But this is failing because it needs at least the empty header file.

**Review by Acuadros95 — APPROVED** [review]

**Pull request merged**